sdk/textanalytics/Azure.AI.TextAnalytics/src/AnalyzeSentimentActionResult.cs

```diff
@@ -13,27 +13,23 @@ public class AnalyzeSentimentActionResult : TextAnalyticsActionResult
     {
         private readonly AnalyzeSentimentResultCollection _documentsResults;
 
-        internal AnalyzeSentimentActionResult(AnalyzeSentimentResultCollection result, DateTimeOffset completedOn, TextAnalyticsErrorInternal error)
-            : base(completedOn, error)
-        {
-            _documentsResults = result;
-        }
-
         /// <summary>
-        /// Intended for mocking purposes only.
+        /// Successful action.
         /// </summary>
-        internal AnalyzeSentimentActionResult(
-            AnalyzeSentimentResultCollection result,
-            DateTimeOffset completedOn) : base(completedOn)
+        internal AnalyzeSentimentActionResult(AnalyzeSentimentResultCollection result, string actionName, DateTimeOffset completedOn)
+            : base(completedOn)
         {
             _documentsResults = result;
+            ActionName = actionName;
         }
 
         /// <summary>
-        /// Intended for mocking purposes only.
+        /// Action with an error.
         /// </summary>
-        internal AnalyzeSentimentActionResult(TextAnalyticsErrorInternal error) : base(error)
+        internal AnalyzeSentimentActionResult(string actionName, DateTimeOffset completedOn, TextAnalyticsErrorInternal error)
+            : base(completedOn, error)
         {
+            ActionName = actionName;
```

Comment:
is the actionName returned when it's an error?

Reply:
it should. ActionName is required in the swagger independent if it worked or not